keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,593 @@
+ConfigMap. The ConfigMap embeds a versioned config.
+
+If the config specifies a registry location, the tool may automatically adjust
+image pushes to push to the specified registry.
+
+If the config specifies a `help` URL, the tool may prompt the user to set up a
+registry for faster development.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- Users may see the ConfigMap and draw mistaken conclusions about how to
+  interact with it. They think deleting the ConfigMap would delete the local
+  registry (which it does not).
+
+- This KEP does not specify any automatic mechanism for keeping the
+  LocalRegistryStatus up-to-date with how the cluster is configured.
+  For example, the user might delete the registry. The cluster might
+  not have a way of knowing that the registry has died.
+  
+- This KEP creates the new API version `local-registry.x-k8s.io/v1alpha1`,
+  This may open a can of worms. On the one hand, cluster implementations want
+  confidence that they're implementing a shared standard. On the other hand,
+  a new API version implies standards of governance, ownership, and API machinery
+  that might not make sense yet.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+None
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Tools that configure a local registry should apply the following ConfigMap to
+the cluster:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-image-loaders
+  namespace: kube-public
+data:
+  registryStatus.v1: |
+    host: "localhost:5000"
+    hostFromCluster: "registry:5000"
+    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
+```
+
+Documentation that educates people on how to set up a registry manually should
+include instructions on how this is applied.
+
+The name of the config map must be `local-image-loaders`.
+
+The namespace must be `kube-public`.
+
+The `registryStatus.v1` field of the ConfigMap data must contain YAML. 
+The following code is the reference implementation that readers may use to parse the YAML:
+
+```go
+
+// A local registry allows clients to load images into the local cluster by
```

Comment:
the export comment should start with `LocalRegistryStatusV1 ...` to comply with golint.

Reply:
done